javalib/src/main/scala/java/lang/Double.scala

```diff
@@ -244,7 +295,13 @@ object Double {
      */
 
     @inline def nativeParseInt(s: String, radix: Int): scala.Double =
-      js.Dynamic.global.parseInt(s, radix).asInstanceOf[scala.Double]
+      linkTimeIf(targetPureWasm) {
+        // java.lang.Long.parseLong may fail with NumberFormatException
+        // for a large input.
+        new BigInteger(s, radix).doubleValue()
+      } {
+        js.Dynamic.global.parseInt(s, radix).asInstanceOf[scala.Double]
+      }
 
     val mantissa = nativeParseInt(truncatedMantissaStr, 16)
```

Comment:
So, concretely, I would leave `nativeParseInt` as being `nativeJSParseInt` and do
```suggestion
    val mantissa = linkTimeIf(targetPureWasm) {
      val mantissaLong = Long.parseUnsignedLong(truncatedMantissaStr, 16)
      // convert unsigned long to double
      (mantissaLong >>> 32).toDouble * (1L << 32) + (mantissaLong & 0xffffffffL).toDouble
    } {
      nativeJSParseInt(truncatedMantissaStr, 16)
    }
```
and below, for the computation of `binaryExp` (which does not show up in the diff so I can't comment on it directly):
```scala
    val binaryExp = linkTimeIf(targetPureWasm) {
      if (binaryExpStr.length() > 11) {
        if (binaryExpStr.charAt(0) = '-') Int.MinValue
        else Int.MaxValue
      } else {
        val binaryExpLong = Long.parseLong(binaryExpStr)
        if (binaryExpLong > Int.MaxValue.toLong) Int.MaxValue
        else if (binaryExpLong < Int.MinValue.toLong) Int.MinValue
        else binaryExpLong.toInt
      }
    } {
      val binaryExpDouble = nativeParseInt(binaryExpStr, 10)
      binaryExpDouble.toInt // caps to [MinValue, MaxValue]
    }
```